### PR TITLE
Implement feature switch to turn PII on/off

### PR DESCRIPTION
**Pull request by antrix1989**
Added PII flag to the Logger only. Right now working on Telementry. Will add it to this PR.

### CODE REVIEW

File: ADAL/src/cache/ADTokenCacheItem+Internal.m [hunk 1]
```diff
@@ -299,13 +285,7 @@ - (void)logMessage:(NSString*)message level:(ADAL_LOG_LEVEL)level correlationId:
...
     }
     
-    [ADLogger log:level
-          context:self
-          message:tokenMessage
-        errorCode:0
-    correlationId:correlationId
-         userInfo:nil
-           format:@"{\n\tresource = %@\n\tclientId = %@\n\tauthority = %@\n\tuserId = %@\n}",
+    [ADLogger log:level context:self errorCode:0 correlationId:correlationId userInfo:nil isPii:YES format:@"%@ {\n\tresource = %@\n\tclientId = %@\n\tauthority = %@\n\tuserId = %@\n}", tokenMessage,
```
Comment (partnerinflight) [inline ADAL/src/cache/ADTokenCacheItem+Internal.m]: MAJOR Split this 199 characters long line (which is greater than 120 authorized). [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3ALineLength)

File: ADAL/src/cache/ADTokenCacheItem+Internal.m [hunk 2]
```diff
@@ -255,23 +255,9 @@ - (void)logMessage:(NSString*)message level:(ADAL_LOG_LEVEL)level correlationId:
...
-        if (message)
-        {
-            tombstoneMessage = [NSString stringWithFormat:@"%@ tombstone : %@", message, _tombstone];
-        }
-        else
-        {
-            tombstoneMessage = [NSString stringWithFormat:@"Tombstone : %@", _tombstone];
-        }
+        [ADLogger log:level context:self errorCode:0 correlationId:correlationId userInfo:nil isPii:YES format:@"%@", _tombstone];
+        [ADLogger log:level context:self errorCode:0 correlationId:correlationId userInfo:nil isPii:NO format:@"{\n\tresource: %@\n\tclientId: %@\n\tauthority:%@\n}", _resource, _clientId, _authority];
```
Comment (partnerinflight) [inline ADAL/src/cache/ADTokenCacheItem+Internal.m]: MAJOR Split this 201 characters long line (which is greater than 120 authorized). [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3ALineLength)

File: ADAL/src/cache/ADTokenCacheItem+Internal.m [hunk 3]
```diff
@@ -40,24 +40,24 @@ @implementation ADTokenCacheItem (Internal)
 - (void)checkCorrelationId:(NSDictionary*)response
       requestCorrelationId:(NSUUID*)requestCorrelationId
 {
-    AD_LOG_VERBOSE(@"Token extraction", requestCorrelationId, @"Attempt to extract the data from the server response.");
+    AD_LOG_VERBOSE(requestCorrelationId, nil, NO, @"Token extraction. Attempt to extract the data from the server response.");
```
Comment (partnerinflight) [inline ADAL/src/cache/ADTokenCacheItem+Internal.m]: MAJOR Split this 126 characters long line (which is greater than 120 authorized). [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3ALineLength)

File: ADAL/src/urlprotocol/ADURLProtocol.m [hunk 4]
```diff
@@ -166,7 +165,8 @@ - (void)startLoading
         _context = context;
     }
     
-    AD_LOG_VERBOSE_F(@"-[ADURLProtocol startLoading]", context.correlationId, @"host: %@", [self.request.URL host]);
+    AD_LOG_VERBOSE(context.correlationId, nil, NO, @"%@ - host: %@", @"-[ADURLProtocol startLoading]", [self.request.URL host]);
```
Comment (partnerinflight) [inline ADAL/src/urlprotocol/ADURLProtocol.m]: MAJOR Split this 128 characters long line (which is greater than 120 authorized). [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3ALineLength)

File: ADAL/src/urlprotocol/ADURLProtocol.m [hunk 5]
```diff
@@ -260,8 +260,8 @@ - (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task
  completionHandler:(ChallengeCompletionHandler)completionHandler
 {
     NSString *authMethod = [challenge.protectionSpace.authenticationMethod lowercaseString];
-    AD_LOG_VERBOSE_F(@"session:task:didReceiveChallenge:completionHandler", _context.correlationId,
-                     @"%@. Previous challenge failure count: %ld", authMethod, (long)challenge.previousFailureCount);
+    
+    AD_LOG_VERBOSE(_context.correlationId, nil, NO, @"%@ - %@. Previous challenge failure count: %ld", @"session:task:didReceiveChallenge:completionHandler", authMethod, (long)challenge.previousFailureCount);
```
Comment (partnerinflight) [inline ADAL/src/urlprotocol/ADURLProtocol.m]: MAJOR Split this 208 characters long line (which is greater than 120 authorized). [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3ALineLength)

File: ADAL/src/cache/ADTokenCacheAccessor.m [hunk 6]
```diff
@@ -402,7 +404,8 @@ - (void)removeImpl:(ADTokenCacheItem *)cacheItem
         return;
     }
     
-    AD_LOG_VERBOSE_F(@"Token cache store", correlationId, @"Tombstoning cache for resource: %@", cacheItem.resource);
+    AD_LOG_VERBOSE(correlationId, nil, NO, @"Token cache store - Tombstoning cache for resource: %@", cacheItem.resource);
```
Comment (partnerinflight) [inline ADAL/src/cache/ADTokenCacheAccessor.m]: MAJOR Split this 122 characters long line (which is greater than 120 authorized). [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3ALineLength)

File: ADAL/src/cache/ADTokenCacheAccessor.m [hunk 7]
```diff
@@ -324,7 +325,8 @@ - (void)updateCacheToItem:(ADTokenCacheItem *)cacheItem
         }
     }
     
-    AD_LOG_VERBOSE_F(@"Token cache store", correlationId, @"Storing access token for resource: %@", cacheItem.resource);
+    AD_LOG_VERBOSE(correlationId, nil, NO, @"Token cache store - Storing access token for resource: %@", cacheItem.resource);
```
Comment (partnerinflight) [inline ADAL/src/cache/ADTokenCacheAccessor.m]: MAJOR Split this 125 characters long line (which is greater than 120 authorized). [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3ALineLength)

**Comment by partnerinflight** [comment]
SonarQube analysis reported 7 issues
* ![MAJOR][MAJOR] 7 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by partnerinflight** [comment]
SonarQube analysis reported 4 issues
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: ADAL/src/ADLogger+Internal.h [hunk 8]
```diff
@@ -23,52 +23,22 @@
...
-#define AD_LOG_WARN_DICT_F(_msg, _correlationId, _dict, _fmt, ...) \
-    AD_LOG_F(ADAL_LOG_LEVEL_WARN, _msg, AD_ERROR_SUCCEEDED, _correlationId, _dict, _fmt, ##__VA_ARGS__)
-#define AD_LOG_INFO_DICT_F(_msg, _correlationId, _dict, _fmt, ...) \
-    AD_LOG_F(ADAL_LOG_LEVEL_INFO, _msg, AD_ERROR_SUCCEEDED, _correlationId, _dict, _fmt, ##__VA_ARGS__)
-#define AD_LOG_VERBOSE_DICT_F(_msg, _correlationId, _dict, _fmt, ...) \
-    AD_LOG_F(ADAL_LOG_LEVEL_VERBOSE, _msg, AD_ERROR_SUCCEEDED, _correlationId, _dict, _fmt, ##__VA_ARGS__)
+#define AD_LOG_ERROR(_code, _correlationId, _dict, _isPii, _fmt, ...) \
+    AD_LOG(ADAL_LOG_LEVEL_ERROR, _code, _correlationId, _dict, _isPii, _fmt, ##__VA_ARGS__)
+
+#define AD_LOG_WARN(_correlationId, _dict, _isPii, _fmt, ...) \
```
Comment (RPangrle) [inline ADAL/src/ADLogger+Internal.h]: If we're going to change these macros that needs some wider design discussion, my preference here would be to:

1. Remove the userInfo parameter entirely. This was a weird hacky way to put in telemetry without really doing telemetry. We have a real telemetry channel now, I'd rather just get rid of this.
2. Instead of isPII being a parameter make separate _PII macros

Comment (RPangrle) [reply ADAL/src/ADLogger+Internal.h]: Also, remove error code from the error logging macros, there's plenty of cases where we want to log error information that's not tied to a specific error code and that's led to teams in the past filing issues based on error log statements rather then actual end result errors.

Comment (oldalton) [reply ADAL/src/ADLogger+Internal.h]: * I agree with number 1 (removing the userInfo and the error code), as it doesn't seem to provide much value. Unless there're good historical reasons to keep it around, I totally support removing them.

* I don't completely agree with having separate _PII macros. From a developer perspective, if I forced to pass the PII flag when using the macro, it forces me to think if it has PII or not. If there's a separate _PII macro instead, I can very well just use the normal macro and miss PII. I'm open to discussion on this one, but that's my personal opinion :)

Comment (antrix1989) [reply ADAL/src/ADLogger+Internal.h]: I agree with Olga's point regarding separate/one macro(s). I think it would be safer to force developer to pass the PII flag.

Comment (unpluggedk) [reply ADAL/src/ADLogger+Internal.h]: I agree with (1) and about removing error code. 
(2), I am open to both; but if we have a separate macro, it will read better in terms of reviews and all vs having a boolean value as one of the parameters.

File: ADAL/src/urlprotocol/ADURLProtocol.m [hunk 9]
```diff
@@ -166,7 +165,8 @@ - (void)startLoading
         _context = context;
     }
     
-    AD_LOG_VERBOSE_F(@"-[ADURLProtocol startLoading]", context.correlationId, @"host: %@", [self.request.URL host]);
+    AD_LOG_VERBOSE(context.correlationId, NO, @"%@ - host: %@", @"-[ADURLProtocol startLoading]", [self.request.URL host]);
```
Comment (partnerinflight) [inline ADAL/src/urlprotocol/ADURLProtocol.m]: MINOR Split this 123 characters long line (which is greater than 120 authorized). [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3ALineLength)

File: ADAL/src/urlprotocol/ADURLProtocol.m [hunk 10]
```diff
@@ -260,8 +260,8 @@ - (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task
  completionHandler:(ChallengeCompletionHandler)completionHandler
 {
     NSString *authMethod = [challenge.protectionSpace.authenticationMethod lowercaseString];
-    AD_LOG_VERBOSE_F(@"session:task:didReceiveChallenge:completionHandler", _context.correlationId,
-                     @"%@. Previous challenge failure count: %ld", authMethod, (long)challenge.previousFailureCount);
+    
+    AD_LOG_VERBOSE(_context.correlationId, NO, @"%@ - %@. Previous challenge failure count: %ld", @"session:task:didReceiveChallenge:completionHandler", authMethod, (long)challenge.previousFailureCount);
```
Comment (partnerinflight) [inline ADAL/src/urlprotocol/ADURLProtocol.m]: MINOR Split this 203 characters long line (which is greater than 120 authorized). [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3ALineLength)

**Comment by partnerinflight** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by partnerinflight** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: ADAL/src/ADAuthenticationError.m [hunk 11]
```diff
@@ -95,19 +95,8 @@ - (id)initInternalWithDomain:(NSString *)domain
...
-            logDict = @{ @"error" : self,
-                         @"correlationId" : correlationId };
-        }
-        else
-        {
-            logDict = @{ @"error" : self };
-        }
-        
-        AD_LOG_ERROR_DICT(message, code, correlationId, logDict, nil);
+        AD_LOG_ERROR(correlationId, @"Error raised: (Domain: \"%@\" Code: %@ ProtocolCode: \"%@\"", domain, codeStr, protocolCode);
```
Comment (oldalton) [inline ADAL/src/ADAuthenticationError.m]: can it be that codeStr has PII depending on the error that gets passed?

File: ADAL/src/ADAuthenticationParameters.m [hunk 12]
```diff
@@ -80,7 +80,7 @@ + (void)parametersFromResourceUrl:(NSURL*)resourceUrl
 
     ADWebRequest* request = [[ADWebRequest alloc] initWithURL:resourceUrl context:nil];
     [request setIsGetRequest:YES];
-    AD_LOG_VERBOSE_F(@"Starting authorization challenge request", nil, @"Resource: %@", resourceUrl);
+    AD_LOG_VERBOSE(nil, @"Starting authorization challenge request. Resource: %@", resourceUrl);
```
Comment (oldalton) [inline ADAL/src/ADAuthenticationParameters.m]: isn't resource OII?

File: ADAL/src/ADLogger.m [hunk 13]
```diff
@@ -309,7 +315,7 @@ + (void)logToken:(NSString *)token
         [logString appendFormat:@" expires on %@", expiresOn];
     }
     
-    AD_LOG_INFO(logString, correlationId, nil);
+    AD_LOG_INFO(correlationId, @"%@", logString);
```
Comment (oldalton) [inline ADAL/src/ADLogger.m]: I think this should be always PII

Comment (antrix1989) [reply ADAL/src/ADLogger.m]: changed

File: ADAL/src/ADAuthenticationParameters.m (hunk 12 above)
Comment (antrix1989) [reply ADAL/src/ADAuthenticationParameters.m]: changed

File: ADAL/src/cache/ADTokenCacheAccessor.m [hunk 14]
```diff
@@ -245,7 +245,7 @@ - (void)updateCacheToResult:(ADAuthenticationResult *)result
            || ![ADAuthenticationContext handleNilOrEmptyAsResult:item argumentName:@"resource" authenticationResult:&result]
            || ![ADAuthenticationContext handleNilOrEmptyAsResult:item argumentName:@"accessToken" authenticationResult:&result])
         {
-            AD_LOG_WARN(@"Told to update cache to an invalid token cache item", [context correlationId], nil);
+            AD_LOG_WARN(nil, @"Told to update cache to an invalid token cache item.");
```
Comment (oldalton) [inline ADAL/src/cache/ADTokenCacheAccessor.m]: why did you remove correlationId logging?

File: ADAL/src/cache/ADTokenCacheAccessor.m [hunk 15]
```diff
@@ -282,7 +282,8 @@ - (void)updateCacheToItem:(ADTokenCacheItem *)cacheItem
     NSString* savedRefreshToken = cacheItem.refreshToken;
     if (isMRRT)
     {
-        AD_LOG_VERBOSE_F(@"Token cache store", correlationId, @"Storing multi-resource refresh token for authority: %@", _authority);
+        AD_LOG_VERBOSE(correlationId, @"Token cache store - Storing multi-resource refresh token for authority: %@", _authority);
+
```
Comment (oldalton) [inline ADAL/src/cache/ADTokenCacheAccessor.m]: can authority be PII?

File: ADAL/src/cache/ADTokenCacheAccessor.m [hunk 16]
```diff
@@ -324,7 +325,8 @@ - (void)updateCacheToItem:(ADTokenCacheItem *)cacheItem
         }
     }
     
-    AD_LOG_VERBOSE_F(@"Token cache store", correlationId, @"Storing access token for resource: %@", cacheItem.resource);
+    AD_LOG_VERBOSE(correlationId, @"Token cache store - Storing access token for resource: %@", cacheItem.resource);
+
```
Comment (oldalton) [inline ADAL/src/cache/ADTokenCacheAccessor.m]: resource is OII

File: ADAL/src/cache/ADTokenCacheAccessor.m [hunk 17]
```diff
@@ -402,7 +404,8 @@ - (void)removeImpl:(ADTokenCacheItem *)cacheItem
         return;
     }
     
-    AD_LOG_VERBOSE_F(@"Token cache store", correlationId, @"Tombstoning cache for resource: %@", cacheItem.resource);
+    AD_LOG_VERBOSE(correlationId, @"Token cache store - Tombstoning cache for resource: %@", cacheItem.resource);
+
```
Comment (oldalton) [inline ADAL/src/cache/ADTokenCacheAccessor.m]: resource is OII

File: ADAL/src/cache/ADTokenCacheItem+Internal.m [hunk 18]
```diff
@@ -255,23 +255,11 @@ - (void)logMessage:(NSString*)message level:(ADAL_LOG_LEVEL)level correlationId:
...
-              context:self
-              message:tombstoneMessage
-            errorCode:0
-        correlationId:correlationId
-             userInfo:_tombstone
+        [ADLogger log:level context:self correlationId:correlationId isPii:YES
+               format:@"%@", _tombstone];
+        [ADLogger log:level context:self correlationId:correlationId isPii:NO
                format:@"{\n\tresource: %@\n\tclientId: %@\n\tauthority:%@\n}", _resource, _clientId, _authority];
+
```
Comment (oldalton) [inline ADAL/src/cache/ADTokenCacheItem+Internal.m]: resource is OII, I believe authority is OII

File: ADAL/src/cache/ios/ADKeychainTokenCache.m [hunk 19]
```diff
@@ -87,12 +87,12 @@ + (void)setDefaultKeychainGroup:(NSString *)keychainGroup
...
     if (s_defaultCache)
     {
-        AD_LOG_ERROR(@"Failed to set default keychain group, default keychain cache has already been instantiated.", AD_ERROR_UNEXPECTED, nil, nil);
+        AD_LOG_ERROR(nil, @"Failed to set default keychain group, default keychain cache has already been instantiated.");
         
         @throw @"Attempting to change the keychain group once AuthenticationContexts have been created or the default keychain cache has been retrieved is invalid. The default keychain group should only be set once for the lifetime of an application.";
     }
     
-    AD_LOG_INFO(([NSString stringWithFormat:@"Setting default keychain group to %@", keychainGroup]), nil, nil);
+    AD_LOG_INFO(nil, @"Setting default keychain group to %@", keychainGroup);
```
Comment (oldalton) [inline ADAL/src/cache/ios/ADKeychainTokenCache.m]: That's an interesting one, can keychain group be OII :)?

File: ADAL/src/cache/ios/ADKeychainTokenCache.m [hunk 20]
```diff
@@ -200,13 +200,12 @@ - (void)logItemRetrievalStatus:(NSArray *)items
     if (!items || [items count]<=0)
     {
         //if resource is nil, this request is intending to find MRRT
-        NSString* logStr = [NSString stringWithFormat:@"No items were found for query: %@", keyCtxStr];
-        AD_LOG_INFO(logStr, correlationId, nil);
+        AD_LOG_INFO(correlationId, @"No items were found for query: %@", keyCtxStr);
```
Comment (oldalton) [inline ADAL/src/cache/ios/ADKeychainTokenCache.m]: Contains resource == OII

File: ADAL/src/cache/ios/ADKeychainTokenCache.m [hunk 21]
```diff
@@ -200,13 +200,12 @@ - (void)logItemRetrievalStatus:(NSArray *)items
...
     }
     else
     {
-        NSString* msg = [NSString stringWithFormat:@"Found %lu token(s) for query: %@", (unsigned long)[items count], keyCtxStr];
-        AD_LOG_INFO_F(msg, correlationId, @"user <%@>", userId);
+        AD_LOG_INFO(correlationId, @"Found %lu token(s) for query: %@", (unsigned long)[items count], keyCtxStr);
```
Comment (oldalton) [inline ADAL/src/cache/ios/ADKeychainTokenCache.m]: see my above comment about query having OII

File: ADAL/src/request/ADAcquireTokenSilentHandler.m [hunk 22]
```diff
@@ -125,7 +125,8 @@ - (void)acquireTokenByRefreshToken:(NSString*)refreshToken
     [[ADWebAuthRequest alloc] initWithURL:[NSURL URLWithString:[[_requestParams authority] stringByAppendingString:OAUTH2_TOKEN_SUFFIX]]
                                   context:_requestParams];
     [webReq setRequestDictionary:request_data];
-    AD_LOG_INFO_F(@"Attempting to acquire an access token from refresh token", nil, @"clientId: '%@'; resource: '%@';", [_requestParams clientId], [_requestParams resource]);
+    
+    AD_LOG_INFO(nil, @"Attempting to acquire an access token from refresh token clientId: '%@'; resource: '%@';", [_requestParams clientId], [_requestParams resource]);
```
Comment (oldalton) [inline ADAL/src/request/ADAcquireTokenSilentHandler.m]: see my above comment about resource being OII

File: ADAL/src/request/ADAcquireTokenSilentHandler.m [hunk 23]
```diff
@@ -231,7 +232,7 @@ - (void)acquireTokenWithItem:(ADTokenCacheItem *)item
              msg = [NSString stringWithFormat:@"Acquire Token with Refresh Token %@.", resultStatus];
          }
          
-         AD_LOG_INFO_F(msg, [_requestParams correlationId], @"clientId: '%@'; resource: '%@';", [_requestParams clientId], [_requestParams resource]);
+         AD_LOG_INFO(_requestParams.correlationId, @"%@ clientId: '%@'; resource: '%@';", msg, _requestParams.clientId, _requestParams.resource);
```
Comment (oldalton) [inline ADAL/src/request/ADAcquireTokenSilentHandler.m]: resource might be OII

File: ADAL/src/request/ADAuthenticationRequest+AcquireAssertion.m [hunk 24]
```diff
@@ -49,7 +49,7 @@ - (void)requestTokenByAssertion:(ADAuthenticationCallback)completionBlock
 {
     [self ensureRequest];
     NSUUID* correlationId = [_requestParams correlationId];
-    AD_LOG_INFO_F(@"Requesting token from SAML Assertion", correlationId, @"resource: %@, clientId: %@", [_requestParams resource], [_requestParams clientId]);
+    AD_LOG_INFO(correlationId, @"Requesting token from SAML Assertion. resource: %@, clientId: %@", _requestParams.resource, _requestParams.clientId);
```
Comment (oldalton) [inline ADAL/src/request/ADAuthenticationRequest+AcquireAssertion.m]: resource might be OII

File: ADAL/src/request/ADAuthenticationRequest+AcquireToken.m [hunk 25]
```diff
@@ -55,25 +55,21 @@ - (void)acquireToken:(NSString *)apiId
...
-    AD_LOG_INFO_F(log, _requestParams.correlationId, @"userId = %@", _requestParams.identifier.userId);
+    AD_LOG_INFO_PII(_requestParams.correlationId, @"userId = %@", _requestParams.identifier.userId);
     
     ADAuthenticationCallback wrappedCallback = ^void(ADAuthenticationResult* result)
     {
-        NSString* finalLog = nil;
         if (result.status == AD_SUCCEEDED)
         {
-            finalLog = [NSString stringWithFormat:@"##### END %@ succeeded. #####", log];
+            AD_LOG_INFO(result.correlationId, @"##### END %@ succeeded. #####", log);
```
Comment (oldalton) [inline ADAL/src/request/ADAuthenticationRequest+AcquireToken.m]: log can have OII through resource and possibly authority, if authority is not a known host

File: ADAL/src/request/ADAuthenticationRequest+AcquireToken.m [hunk 26]
```diff
@@ -55,25 +55,21 @@ - (void)acquireToken:(NSString *)apiId
...
         }
         else
         {
             ADAuthenticationError* error = result.error;
-            finalLog = [NSString stringWithFormat:@"##### END %@ failed { domain: %@ code: %ld protocolCode: %@ errorDetails: %@} #####",
-                        log, error.domain, (long)error.code, error.protocolCode, error.errorDetails];
+            AD_LOG_INFO(result.correlationId, @"##### END %@ failed { domain: %@ code: %ld protocolCode: %@ ", log, error.domain, (long)error.code, error.protocolCode);
```
Comment (oldalton) [inline ADAL/src/request/ADAuthenticationRequest+AcquireToken.m]: see above comment about log

File: ADAL/src/request/ADAuthenticationRequest+AcquireToken.m [hunk 27]
```diff
@@ -461,7 +457,8 @@ - (void)requestTokenByCode:(NSString *)code
 {
     HANDLE_ARGUMENT(code, [_requestParams correlationId]);
     [self ensureRequest];
-    AD_LOG_VERBOSE_F(@"Requesting token from authorization code.", [_requestParams correlationId], @"Requesting token by authorization code for resource: %@", [_requestParams resource]);
+    
+    AD_LOG_VERBOSE(_requestParams.correlationId, @"Requesting token by authorization code for resource: %@", _requestParams.resource);
```
Comment (oldalton) [inline ADAL/src/request/ADAuthenticationRequest+AcquireToken.m]: resource might be OII

File: ADAL/src/request/ADAuthenticationRequest+WebRequest.m [hunk 28]
```diff
@@ -85,11 +85,12 @@ - (BOOL)verifyStateFromDictionary: (NSDictionary*) dictionary
         
         if (![NSString adIsStringNilOrBlank:authorizationServer] && ![NSString adIsStringNilOrBlank:resource])
         {
-            AD_LOG_VERBOSE_F(@"State", [_requestParams correlationId], @"The authorization server returned the following state: %@", state);
+            AD_LOG_VERBOSE(_requestParams.correlationId, @"The authorization server returned the following state: %@", state);
```
Comment (oldalton) [inline ADAL/src/request/ADAuthenticationRequest+WebRequest.m]: state is base64 encoded parameters, which can have resource and authorization server

File: ADAL/src/request/ADAuthenticationRequest+WebRequest.m [hunk 29]
```diff
@@ -85,11 +85,12 @@ - (BOOL)verifyStateFromDictionary: (NSDictionary*) dictionary
...
             return YES;
         }
     }
-    AD_LOG_WARN_F(@"State error", [_requestParams correlationId], @"Missing or invalid state returned: %@", state);
+    
+    AD_LOG_WARN(_requestParams.correlationId, @"Missing or invalid state returned: %@", state);
```
Comment (oldalton) [inline ADAL/src/request/ADAuthenticationRequest+WebRequest.m]: state should be treated carefully

File: ADAL/src/request/ADAuthenticationRequest+WebRequest.m [hunk 30]
```diff
@@ -174,7 +175,7 @@ - (void)requestCode:(ADAuthorizationCodeCallback)completionBlock
     THROW_ON_NIL_ARGUMENT(completionBlock);
     [self ensureRequest];
     
-    AD_LOG_VERBOSE_F(@"Requesting authorization code.", _requestParams.correlationId, @"Requesting authorization code for resource: %@", _requestParams.resource);
+    AD_LOG_VERBOSE(_requestParams.correlationId, @"Requesting authorization code for resource: %@", _requestParams.resource);
```
Comment (oldalton) [inline ADAL/src/request/ADAuthenticationRequest+WebRequest.m]: resource might be OII

File: ADAL/src/ui/ADWebAuthController.m [hunk 31]
```diff
@@ -250,14 +250,16 @@ - (void)webAuthDidStartLoad:(NSURL*)url
 
 - (void)webAuthDidFinishLoad:(NSURL*)url
 {
-    AD_LOG_VERBOSE_F(@"-webAuthDidFinishLoad:", _requestParams.correlationId, @"host: %@", url.host);
+    AD_LOG_VERBOSE(_requestParams.correlationId, @"-webAuthDidFinishLoad host: %@", url.host);
```
Comment (oldalton) [inline ADAL/src/ui/ADWebAuthController.m]: depending on the url, it might be OII

File: ADAL/src/urlprotocol/ADNTLMHandler.m [hunk 32]
```diff
@@ -85,7 +85,7 @@ + (BOOL)handleChallenge:(NSURLAuthenticationChallenge *)challenge
         }
         
         // This is the NTLM challenge: use the identity to authenticate:
-        AD_LOG_INFO_F(@"Attempting to handle NTLM challenge", nil,  @"host: %@", challenge.protectionSpace.host);
+        AD_LOG_INFO(nil, @"Attempting to handle NTLM challenge - host: %@", challenge.protectionSpace.host);
```
Comment (oldalton) [inline ADAL/src/urlprotocol/ADNTLMHandler.m]: host can be OII

File: ADAL/src/utils/ADHelpers.m [hunk 33]
```diff
@@ -327,13 +327,13 @@ + (NSString*)canonicalizeAuthority:(NSString *)authority
     NSURL* url = [NSURL URLWithString:trimmedAuthority];
     if (!url)
     {
-        AD_LOG_WARN_F(@"The authority is not a valid URL", nil, @"Authority %@", authority);
+        AD_LOG_WARN(nil, @"The authority is not a valid URL %@", authority);
```
Comment (oldalton) [inline ADAL/src/utils/ADHelpers.m]: authority can be OII

File: ADAL/src/utils/ios/ADKeychainUtil.m [hunk 34]
```diff
@@ -36,7 +36,7 @@ + (NSString*)keychainTeamId:(ADAuthenticationError* __autoreleasing *)error
         ADAuthenticationError* localError = nil;
         s_keychainTeamId = [self retrieveTeamIDFromKeychain:&localError];
         adError = localError;
-        AD_LOG_INFO(([NSString stringWithFormat:@"Using \"%@\" Team ID for Keychain.", s_keychainTeamId]), nil, nil);
+        AD_LOG_INFO(nil, @"Using \"%@\" Team ID for Keychain.", s_keychainTeamId);
```
Comment (oldalton) [inline ADAL/src/utils/ios/ADKeychainUtil.m]: keychain team ID can be OII?

File: ADAL/src/workplacejoin/ADJwtHelper.m [hunk 35]
```diff
@@ -144,7 +144,8 @@ + (NSString *)JSONFromDictionary:(NSDictionary *)dictionary
                                                          error:&error];
     if (!jsonData)
     {
-        AD_LOG_ERROR(([NSString stringWithFormat:@"Got an error: %@", error]), error.code, nil, nil);
+        AD_LOG_ERROR_PII(nil, @"Got an error: %@", error);
```
Comment (oldalton) [inline ADAL/src/workplacejoin/ADJwtHelper.m]: should we log a non PII version here?

File: ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m [hunk 36]
```diff
@@ -92,11 +92,10 @@ + (ADRegistrationInformation *)getRegistrationInformation:(id<ADRequestContext>)
         {
             *error = adError;
         }
-        AD_LOG_ERROR_F(@"Failed to retrieve WPJ private key reference - ", adError.code, nil, @"%@ correlation id.", context.correlationId);
+        AD_LOG_ERROR(context.correlationId, NO, @"Failed to retrieve WPJ private key reference. Error code %ld", (long)adError.code);
```
Comment (oldalton) [inline ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m]: Why logger calls here still pass the PII boolean flag?

File: ADAL/src/cache/ADTokenCacheAccessor.m (hunk 14 above)
Comment (antrix1989) [reply ADAL/src/cache/ADTokenCacheAccessor.m]: fixed
